src/ui/Combobox/ComboboxValues.ts

```diff
@@ -0,0 +1,189 @@
+
+    if (!targetElement) {
+      return;
+    }
+
+    const value = find(this.values, ({ element }) => element.getAttribute('id') === targetElement.getAttribute('id'));
+    value && this.combobox.options.onSelectValue(value);
+    this.combobox.clearAll();
+  }
+
+  private updateAccessibilityAttributes() {
+    const activeDescendant = this.keyboardActiveValue ? this.keyboardActiveValue.element.getAttribute('id') : '';
+
+    this.combobox.updateAccessibilityAttributes({
+      activeDescendant,
+      expanded: this.hasValues()
+    });
+  }
+
+  public clearValues() {
+    this.mouseIsOverValue = false;
+    this.resetKeyboardActiveValue();
+    $$(this.element).empty();
+    $$(this.element).hide();
+    this.values = [];
+    this.updateAccessibilityAttributes();
+  }
+
+  private setKeyboardActiveValue(value: IComboboxValue) {
+    this.keyboardActiveValue = value;
+    this.activateFocusOnValue(this.keyboardActiveValue);
+  }
+
+  private resetKeyboardActiveValue() {
+    if (!this.keyboardActiveValue) {
+      return;
+    }
+
+    this.deactivateFocusOnValue(this.keyboardActiveValue);
+    this.keyboardActiveValue = null;
+  }
+
+  private activateFocusOnValue({ element }: IComboboxValue) {
+    $$(element).addClass('coveo-focused');
+    element.setAttribute('aria-selected', 'true');
+  }
+
+  private deactivateFocusOnValue({ element }: IComboboxValue) {
+    $$(element).removeClass('coveo-focused');
+    element.setAttribute('aria-selected', 'false');
+  }
+
+  public selectActiveValue() {
+    if (!this.keyboardActiveValue) {
+      return;
+    }
+
+    this.combobox.options.onSelectValue(this.keyboardActiveValue);
+    this.combobox.clearAll();
+  }
+
+  public moveActiveValueDown() {
+    if (!this.hasValues()) {
+      return;
+    }
+
+    const nextActiveValue = this.nextOrFirstValue;
+    this.resetKeyboardActiveValue();
+    this.setKeyboardActiveValue(nextActiveValue);
+    this.updateAccessibilityAttributes();
+  }
+
+  public moveActiveValueUp() {
+    if (!this.hasValues()) {
+      return;
+    }
+
+    const previousActiveValue = this.previousOrLastValue;
+    this.resetKeyboardActiveValue();
+    this.setKeyboardActiveValue(previousActiveValue);
+    this.updateAccessibilityAttributes();
+  }
+
+  private get nextOrFirstValue() {
+    if (!this.keyboardActiveValue) {
+      return this.values[0];
+    }
+
+    const nextValueIndex = this.values.indexOf(this.keyboardActiveValue) + 1;
+    return nextValueIndex < this.values.length ? this.values[nextValueIndex] : this.values[0];
+  }
+
+  private get previousOrLastValue() {
+    const lastValueIndex = this.values.length - 1;
+    if (!this.keyboardActiveValue) {
+      return this.values[lastValueIndex];
+    }
+
+    const previousValueIndex = this.values.indexOf(this.keyboardActiveValue) - 1;
+    return previousValueIndex >= 0 ? this.values[previousValueIndex] : this.values[lastValueIndex];
```

Comment:
same here
```
const nextValueIndex = (this.values.indexOf(this.keyboardActiveValue) - 1) %  this.values.length;
return this.values[nextValueIndex]
```

Reply:
It's the same code right?

Reply:
Ah non just saw the `-` 😉

Reply:
Don't think this works though 😢

Reply:
Could you explain why you don't think it will work? 
Because if we say we have `this.values.length = 5`
```
-1 % 5 = 4
0 % 5 = 0
1 % 5 = 1
2 % 5 = 2
3 % 5 = 3
4 % 5 = 4
```
I don't see where it will not work.

Reply:
Must be because in Javascript % is not really a modulo but a remainder operator, so -1 % 5 = -1

Reply:
ok if we want to have the same code for next and previous, then you could add `this.values.length` like so

`const nextValueIndex = (this.values.indexOf(this.keyboardActiveValue) + this.values.length - 1) %  this.values.length;`